packages/@react-stately/utils/test/number.test.ts

```diff
@@ -8,6 +8,30 @@ describe('number utils', () => {
     });
   });
 
+  describe('roundToStepPrecision', () => {
+    it('should return the input unchanged for integer steps', () => {
+      expect(roundToStepPrecision(7.123, 1)).toBe(7.123);
+      expect(roundToStepPrecision(5, 10)).toBe(5);
+    });
+    it('should round to the correct decimal places for steps with decimals', () => {
+      expect(roundToStepPrecision(1.24, 0.1)).toBe(1.24);
```

Comment:
maybe I'm over thinking this, but shouldn't this be
```suggestion
      expect(roundToStepPrecision(1.24, 0.1)).toBe(1.2);
```

Reply:
hi @scrgl thanks for pointing out that incorrect implementation! https://github.com/adobe/react-spectrum/pull/8290/commits/872c52702bae55fa43008b23976ae0d0503e6f41 fixed the issue mentioned in https://github.com/adobe/react-spectrum/pull/8290#issuecomment-2929973595 and added more tests.

I opted not to use `split` as suggested in the snippet because it creates additional intermediate arrays every time `snapValueToStep` is called. Since this function may be invoked multiple times by hooks and components, those arrays could quickly add up. Also, since `step` values without explicit exponential notation are likely more common in user land, normalizing everything to exponential format upfront would add extra overhead for those cases.